packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -4361,6 +4404,21 @@ abstract class Element extends DiagnosticableTree implements BuildContext {
     return chain;
   }
 
+  @override
+  void dispatchNotification(Notification notification) {
+    _notificationTree?.dispatchNotification(notification);
+  }
+
+  @override
+  bool onNotification(Notification notification) {
+    return false;
+  }
+
+  /// Whether this element can handle [Notifications].
+  ///
+  /// The value of this getter must be unchanging for the lifetime of the Element.
+  bool get handlesNotification => false;
+
```

Comment:
@goderbauer this is basically the new API I added to make this work. I'm not 100% how I feel about it, but it does allow the viewport elements to just mixin a class to support the scroll notifications

Reply:
You mentioned you tried mixin's, but could we have something like this:

* `Element._attachNotificationTree` always sets `_notificationTree = _parent?._notificationTree;`
* a new `NotifiableElementMixin` overrides `_attachNotificationTree` to set itself as the `_notificationTree` and it provides the abstract `onNotification` method.
* `Element.onNotification` is removed

That way we wouldn't need the awkward `handlesNotification` thing...

Users would just create custom elements with `NotifiableElementMixin` mixed in and implement the onNotification method.

Reply:
You can't apply a mixin to a type that overrides a private method. For example, the _attachNotificationTree override defined by the mixin will conflict with the _attachNotificationTree that the NotificationElement inherits from Element.

The only way around this is to make the method public. We could use a protected annotation though. I do like this API better than what I have.

Reply:
> You can't apply a mixin to a type that overrides a private method.

This seems to work in dartpad:

```dart
class Foo {
  foo() {
    _attachNotificationTree();
  }
  
  _attachNotificationTree() {
    print("hello");
  }
}

mixin FooMixin on Foo {
  @override
  _attachNotificationTree() {
    print("Hell from mixin");
  }
}

class Ha extends Foo with FooMixin {
  
}

main() {
  Ha().foo();
}
```

Reply:
Ugh, never mind. This only works because Ha is in the same file...

Reply:
Making it protected doesn't sound like the worst idea then...

Reply:
Yeah, its not so bad - and its not the only element API that is protected too